lib/lib-table-cluster/src/test/java/org/sagebionetworks/table/cluster/TableAndColumnMapperTest.java

```diff
@@ -80,7 +86,76 @@ public void testConstructorWithNullProvider() throws ParseException {
 		}).getMessage();
 		assertEquals("SchemaProvider is required.", message);
 	}
+	
+	@Test
+	public void testConstructorWithEmptySchema() throws ParseException {
+		QuerySpecification model = new TableQueryParser("select * from syn123").querySpecification();
+		Map<IdAndVersion, List<ColumnModel>> map = new LinkedHashMap<>();
+		map.put(IdAndVersion.parse("syn123"), Collections.emptyList());
+		
+		String message = assertThrows(IllegalArgumentException.class, ()->{
+			// call under test
+			new TableAndColumnMapper(model, createSchemaProvider(map));
+		}).getMessage();
+		assertEquals("Schema for syn123 is empty.", message);
+	}
+	
+	@Test
+	public void testBuildSelectAllColumns() throws ParseException {
+		QuerySpecification model = new TableQueryParser("select * from syn123").querySpecification();
+		Map<IdAndVersion, List<ColumnModel>> map = new LinkedHashMap<>();
+		map.put(IdAndVersion.parse("syn123"), allColumns);
+		TableAndColumnMapper mapper = new TableAndColumnMapper(model, createSchemaProvider(map));
+		
+		// call under test
+		SelectList selectList = mapper.buildSelectAllColumns();
+		assertNotNull(selectList);
+		assertEquals("\"foo\", \"has space\", \"bar\", \"foo_bar\", \"Foo\", \"datetype\", \"has\"\"quote\"", selectList.toSql());
+	}
 
+	
+	@Test
+	public void testBuildSelectAllColumnsWithJoinWithoutAlias() throws ParseException {
+		QuerySpecification model = new TableQueryParser("select * from syn123 join syn456").querySpecification();
+		Map<IdAndVersion, List<ColumnModel>> map = new LinkedHashMap<>();
+		map.put(IdAndVersion.parse("syn123"), Arrays.asList(allColumns.get(0), allColumns.get(1)));
+		map.put(IdAndVersion.parse("syn456"), Arrays.asList(allColumns.get(2), allColumns.get(3)));
+		TableAndColumnMapper mapper = new TableAndColumnMapper(model, createSchemaProvider(map));
+		
+		// call under test
+		SelectList selectList = mapper.buildSelectAllColumns();
+		assertNotNull(selectList);
+		assertEquals("syn123.\"foo\", syn123.\"has space\", syn456.\"bar\", syn456.\"foo_bar\"", selectList.toSql());
+	}
+	
+	@Test
+	public void testBuildSelectAllColumnsWithJoinWithAlias() throws ParseException {
+		QuerySpecification model = new TableQueryParser("select * from syn123 t join syn456 r").querySpecification();
+		Map<IdAndVersion, List<ColumnModel>> map = new LinkedHashMap<>();
+		map.put(IdAndVersion.parse("syn123"), Arrays.asList(allColumns.get(0), allColumns.get(1)));
+		map.put(IdAndVersion.parse("syn456"), Arrays.asList(allColumns.get(2), allColumns.get(3)));
+		TableAndColumnMapper mapper = new TableAndColumnMapper(model, createSchemaProvider(map));
+		
+		// call under test
+		SelectList selectList = mapper.buildSelectAllColumns();
+		assertNotNull(selectList);
+		assertEquals("t.\"foo\", t.\"has space\", r.\"bar\", r.\"foo_bar\"", selectList.toSql());
```

Comment:
Should be: assertEquals("\`t\`.\"foo\", \"\`t\`.\"has space\", \`r\`.\"bar\", \`r\`.\"foo_bar\"", selectList.toSql());

Reply:
Ideally we should probably use the ` also for the column names.